docs/dev/table/functions.md

```diff
@@ -1274,6 +1274,17 @@ ATAN2(numeric1, numeric2)
       </td>
     </tr>
 
+    <tr>
+      <td>
+        {% highlight text %}
+COSH(numeric)
+{% endhighlight %}
+      </td>
+      <td>
+        <p>Returns the hyperbolic cosine of <i>numeric</i>. Return value type is DOUBLE.</p>
```

Comment:
Could you make type references consistent `<i>TYPE_NAME</i>? You are using any combination of small case, capital case, with or without italic :)

Reply:
@pnowojski updated `double` and `DOUBLE` with `<i>double</i>`. It's OK?

Reply:
Changed~

Reply:
1. This page is not a pure markdown doc. We cannot use backquote here.
2. Conventionally, we use small letters for SQL attributes and capital letters for Java/Scala attributes.
3. I don't think the return type should be marked like that. In most cases, it should be identical with the input's types.

Reply:
1. in that case I would suggest <i>DOUBLE</i> and <i>NUMERIC</i>. @yanghua could you make sure that the doc page renders correctly (`./build_docs.sh -i` script in `docs` directory allows you to run the docs locally). 
2. https://ci.apache.org/projects/flink/flink-docs-master/dev/table/sql.html#data-types here the types are in capital letter, however I don't mind either way.
3. Unfortunately this is either hard, or impossible:
  a) some functions can not be implemented returning accurate `DECIMAL`, like most of the math functions (`cos`, `sqrt`, `log`, `ln`, ....). 
  b) some function require extra time and effort. Like specialised version of `power(DECIMAL, INT)` could return precise `DECIMAL`, while `power(INT, DECIMAL)` can not.

I think the (most?) popular convention in SQL is to return `DOUBLE` if the result is not precise, to avoid silent precision losses. But correct `DECIMAL` handling is generally speaking bigger topic to sort out in Flink in the future. For example I do not like that we provide functions like `DOUBLE COS(DECIMAL)`, because this also gives an opportunity to semi silent precision losses.

However regardless of that, letting user know whether the function returns precise or approximate result is must have (vide non obvious case of `POWER(2.5, 2)`). Also letting user know what's the return type of the function is also important.

Reply:
@pnowojski  It can not be parsed accurately, it shows : 

```
Returns the hyperbolic cosine of `NUMERIC`. Return value type is `DOUBLE`.
```

Reply:
You can use `<code>NUMERIC</code>` to achieve the same result. It is already widely used in this same file.

Reply:
@pnowojski changed, `<code></code>` works fine.

Reply:
Besides, now that 
> the (most?) popular convention in SQL is to return DOUBLE if the result is not precise

why still indicating the return type for each numeric function explicitly? How about adding a single note for that?

Reply:
@xccui if you think using `<i>` is better I'm fine with that, I don't mind one way or the other.

However I don't understand:

> before reaching a consensus about the format and the internal implementation, I suggest to keep the current doc style, i.e., without mentioning the return type, for now.

We are already following a convention that for all of the math functions we are dropping precision and returning `double`. What's the point of not documenting it?

If you could guarantee me that we will document those return types lets say within a month, I would be fine with postponing it. However I don't believe something like that would happen, so I prefer to incrementally start adding this return type information right now. Later if someone will want, we can reformat the page. I do not see any downside to such incremental approach until that moment.

> why still indicating the return type for each numeric function explicitly? How about adding a single note for that?

since some of them can be implemented precisely (`abs`, `min`, `max`, `ceil`, ...), others can be for some variants (`power(DECIMAL, INT)`). We can state this general rule that functions returning `DOUBLE` are approximations, but we still need specific information for every function, whether it returns approximated (`DOUBLE`) or precise (`DECIMAL`) result.

Reply:
Hi @pnowojski, yes, it's necessary to mark the data types of the functions' inputs/output, but maybe not inline with the description. I think it's better to assign separate columns for that. Anyway, incrementally adding the return types is not bad to me and we need to go over the whole function list at a time in the future.